includes/Feed/FeedFileUploadHandler.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Feed;
+
+use SkyVerge\WooCommerce\PluginFramework\v5_10_0 as Framework;
+
+
+defined( 'ABSPATH' ) || exit;
+/**
+ * Class FeedFileUploadHandler
+ *
+ * @since 2.6.0
+ */
+class FeedFileUploadHandler {
+
+	/**
+	 * Feed file handling and manipulation class.
+	 *
+	 * @var FeedFileHandler
+	 */
+	protected $feed_file_handler;
+
+
+	/**
+	 * Singleton instance.
+	 *
+	 * @var FeedFileUploadHandler
+	 */
+	protected static $instance;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param FeedFileHandler $feed_file_handler Feed file handling.
+	 */
+	public function __construct( $feed_file_handler ) {
+		$this->feed_file_handler = $feed_file_handler;
+		add_action( 'woocommerce_api_' . $feed_file_handler::REQUEST_FEED_ACTION, array( $this, 'handle_feed_request' ) );
+	}
+
+	/**
+	 * Initialize the class.
+	 *
+	 * @since 2.6.0
+	 * @param FeedFileHandler $feed_file_handler Feed file handling.
+	 * @return void
+	 */
+	public static function init( $feed_file_handler ) {
+		if ( null === self::$instance ) {
+			self::$instance = new self( $feed_file_handler );
+		}
+	}
+
+	/**
+	 * Handles the feed data request.
+	 *
+	 * @since 2.6.0
+	 * @throws Framework\SV_WC_Plugin_Exception Feed request not possible.
+	 */
+	public function handle_feed_request() {
+
+		$file_path = $this->feed_file_handler->get_file_path();
+
+		try {
+
+			// Bail early if the feed secret is not included or is not valid.
+			if ( $this->feed_file_handler->get_feed_secret() !== Framework\SV_WC_Helper::get_requested_value( 'secret' ) ) {
```

Comment:
If it would be only used by this class it would make sense to do so, but it will be used in one additional place when we will create the feed file configuration in Facebook Catalog datasources.